pkg/config/sink.go

```diff
@@ -155,7 +155,18 @@ func (s *SinkConfig) applyParameter(sinkURI *url.URL) error {
 		return cerror.ErrSinkURIInvalid.GenWithStackByArgs(errMsg)
 	}
 
-	s.Protocol = params.Get(ProtocolKey)
+	protocolFromURI := params.Get(ProtocolKey)
+	if protocolFromURI != "" {
+		if s.Protocol != "" {
+			log.Warn(
```

Comment:
Do you think it is better to prohibit  behavior and only allow user to specify `protocol` by one means in the same time? 
I also prefer this, but it would breaks compatibility.